server/datastore/mysql/mysql.go

```diff
@@ -283,8 +287,23 @@ func (ds *Datastore) writeChanLoop() {
 	}
 }
 
+var tracedDriverName string
+
+func init() {
```

Comment:
Why not execute this code block in 305 instead of in an `init` func:
```go
if opts.tracingEnabled {
   // ... here ...
   driverName = tracedDriverName
}
```

Reply:
This prevents multipule sql driver registers. Where I had it before around there, it was getting called each time a connection in the pool was being made which caused issues.

Reply:
Ah ok, maybe using `sync.Once` would make sense here.

Just worried about any side effects of always running this code (even if `config.Logging.TracingEnabled` when `false`)

Reply:
No side effects, in fact, the upstream library does it in an init() function itself.

Reply:
https://github.com/elastic/apm-agent-go/blob/main/module/apmsql/mysql/init.go